File: android/LiveData/app/src/main/java/com/example/livedata/MainActivity.java
package com.example.livedata;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;

import com.example.livedata.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
  private ActivityMainBinding binding;
  private MainViewModel model;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    binding = ActivityMainBinding.inflate(getLayoutInflater());
    View view = binding.getRoot();
    setContentView(view);

    model = new ViewModelProvider(this).get(MainViewModel.class);
    model.getCount().observe(this, new Observer<Integer>() {
      @Override
      public void onChanged(@Nullable final Integer count) {
        binding.textView.setText(String.valueOf(count));
      }
    });

    binding.button.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        model.increase(1);
      }
    });
  }
}

File: android/FirstKtApp/app/src/main/java/com/example/firstktapp/MainActivity.kt
package com.example.firstktapp

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 直接調用綁定
        R.id.hello.onClick(this) {
            println("click hello")
        }
    }
}

// 擴展 Int 函數，讓 id 可以直接調用 onClick 方法綁定事件
fun Int.onClick(activity: Activity, onClick: (view: View) -> Unit) {
    activity.findViewById<View>(this).setOnClickListener {
        println("Int.onClick")
        onClick(it)
    }
}